### ISSUE TITLE
Support 204 No Content in asJson[Unit]?

### ISSUE THREAD

**Issue by nightscape** [softwaremill/sttp#675]
Hi all,

I've auto-created an STTP client for the Azure Digital Twins API using https://github.com/OpenAPITools/sbt-openapi-generator on their OpenAPI specs.
The generator created `.response(asJson[Unit])` calls for `204 No Content` responses.
Similar to https://github.com/softwaremill/sttp/issues/161 this creates a parsing error because of the empty body.
Would it be reasonable to extend the existing `asJson[Option[T]]` functionality to handle `Unit` by returning `()`?

**Comment by nightscape** [comment]
I could imagine generalizing the `IsOption` typeclass to something like this:
```scala
trait CanParseEmptyString[T] {
  def valueForEmptyString: T
}
```
If `valueForEmptyString` is `t` it would short-circuit the entire JSON parsing and return `t`.

**Comment by adamw** [comment]
Or maybe that's something to fix in the generator? I'd imagine the body for the `204` case being marked as optional, which should result in at least a `asJson[Option[Unit]]`, or even better, special-casing for an optional empty body so that the request is generated with `.ignore`?

**Comment by vakhtang** [comment]
I'm also running into this issue. Using `ignore` isn't an option in my case because the response body contains an error message if the request wasn't successful. `asJson[Option[Unit]]` works but doesn't make a whole lot of sense semantically. Is there any plan on supporting `asJson[Unit]`? I should note that for older versions of sttp `asJson[Unit]` worked without issue.

**Comment by adamw** [comment]
@vakhtang wouldn't `asEither(asString, ignore)` work for your case? Then, `2xx` responses are ignored, and all others are represented as a string

You could also create more condition branches if needed, using `fromMetadata`

**Comment by vakhtang** [comment]
@adamw That would work, but it would require updating the template and I was hoping I wouldn't have to. I'll bite the bullet though.

**Comment by adamw** [comment]
@vakhtang Well I guess it would be best to fix the root cause in the generator :) There's also an alternative one - https://github.com/ghostbuster91/sttp-openapi-generator - maybe that would work for you.

**Comment by Pask423** [comment]
Seem to be fix in `6.2.0` of `openapi-generator`.
